osutil/io.go

```diff
@@ -41,16 +43,50 @@ const (
 // all unit tests in genreal.
 var snapdUnsafeIO bool = len(os.Args) > 0 && strings.HasSuffix(os.Args[0], ".test") && GetenvBool("SNAPD_UNSAFE_IO")
 
-// AtomicWriteFile updates the filename atomically and works otherwise
-// like io/ioutil.WriteFile()
-//
-// Note that it won't follow symlinks and will replace existing symlinks
-// with the real file
-func AtomicWriteFile(filename string, data []byte, perm os.FileMode, flags AtomicWriteFlags) (err error) {
-	return AtomicWriteFileChown(filename, data, perm, flags, -1, -1)
+// An AtomicWriter is an io.WriteCloser that has a Finalize() method that does
+// whatever needs to be done so the edition is "atomic": an AtomicWriter will
+// do its best to leave either the previous content or the new content in
+// permanent storage. It also has a Cancel() method to abort and clean up.
+type AtomicWriter interface {
+	io.WriteCloser
+
+	// Finalize the writing operation and make it permanent.
+	//
+	// If Finalize succeeds, the file is closed and further attempts to write will
+	// fail. If Finalize fails, Cancel() needs to be called to clean up.
+	Finalize() error
```

Comment:
not a fan of this term,   because go has  finalizers that are about gc vs resource freeing

Reply:
fwiw we use Commit already in the code base even for stuff that can be called only once